packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/feature_items_list/blocs/items_list_bloc.dart

```diff
@@ -0,0 +1,56 @@
+{{> licence.dart }}
+
+import 'package:rx_bloc/rx_bloc.dart';
+import 'package:rxdart/rxdart.dart';
+
+import '../../base/common_services/item_service.dart';
+import '../../base/extensions/error_model_extensions.dart';
+import '../../base/models/errors/error_model.dart';
+import '../../base/models/item_model.dart';
+
+part 'items_list_bloc.rxb.g.dart';
+
+/// A contract class containing all events of the ItemsListBloC.
+abstract class ItemsListBlocEvents {
+  void getItemsList();
+}
+
+/// A contract class containing all states of the ItemsListBloC.
+abstract class ItemsListBlocStates {
+  /// The loading state
+  Stream<bool> get isLoading;
+
+  /// The error state
+  Stream<ErrorModel> get errors;
+
+  Stream<Result<List<ItemModel>>> get itemsList;
+}
+
+@RxBloc()
+class ItemsListBloc extends $ItemsListBloc {
+  ItemsListBloc(ItemService itemService) {
+    _$getItemsListEvent
+        .startWith(null)
+        .switchMap((_) => itemService.fetchItems().asResultStream())
+        .bind(_itemsListSubject)
+        .addTo(_compositeSubscription);
+  }
+
+  final _itemsListSubject =
+  BehaviorSubject<Result<List<ItemModel>>>.seeded(Result.loading());
```

Comment:
Please avoid creating subjects within the BloCs unless you really don'y need to. For example we need to create subjects if we need to implement local result filtering

Reply:
Reworked